content/_code-samples/key-derivation/js/index.js

```diff
@@ -0,0 +1,219 @@
+        // https://github.com/indutny/elliptic/blob/master/lib/elliptic/curves.js#L182
+        // How often will an (essentially) random number generated by Sha512 be larger than that?
+        // There's 2^32 chances (the for loop) to get a number smaller than the order,
+        // and it's rare that you'll even get past the first loop iteration.
+        // Note that in TypeScript we actually need the throw, otherwise the function signature would be BN | undefined
+        //
+        /* istanbul ignore next */
+        throw new Error('impossible unicorn ;)');
+    }
+
+    getBase58ED25519Account() {
+        const buffer = Buffer.from(this.ED25519Keypair.publicKey, "hex")
+        return encodeAccountPublic([...buffer])
+    }
+
+    getBase58ESecp256k1Account(validator = false) {
+        const buffer = Buffer.from(this.Secp256K1Keypair.publicKey, "hex")
+        if (validator) {
+            return encodeNodePublic([...buffer])
+        } else {
+            return encodeAccountPublic([...buffer])
+        }
+    }
+
+    toString() {
+        const base58Seed = codec.encodeChecked([XRPL_SEED_PREFIX].concat(this.bytes))
+        const hexSeed = Buffer.from(this.bytes).toString('hex').toUpperCase()
+        const rfc1751Rippled = keyToRFC1751Mnemonic(bytesToHex(this.bytes))
+
+        const base58ED25519Account = this.getBase58ED25519Account();
+
+        const base58Secp256k1Account = this.getBase58ESecp256k1Account();
+        const base58Secp256k1AccountValidator = this.getBase58ESecp256k1Account(true);
+
+        const out = `
+            Seed (base58): ${base58Seed}
+            Seed (hex): ${hexSeed}
+            Seed (rippled RFC-1751): ${rfc1751Rippled}
+            
+            Ed25519 Secret Key (hex): ${this.ED25519Keypair.privateKey}
+            Ed25519 Public Key (hex): ${this.ED25519Keypair.publicKey}
+            Ed25519 Public Key (base58 - Account): ${base58ED25519Account}
+            
+            secp256k1 Secret Key (hex): ${this.Secp256K1Keypair.privateKey}
+            secp256k1 Public Key (hex): ${this.Secp256K1Keypair.publicKey}
+            secp256k1 Public Key (base58 - Account): ${base58Secp256k1Account}
+            secp256k1 Public Key (base58 - Validator): ${base58Secp256k1AccountValidator}
+        `
+
+        return out
+    }
+}
+
+if (process.argv[2] !== undefined) {
+    console.log(process.argv[2]);
+    const seed = new Seed(process.argv[2]);
+    console.log(seed.toString())
+} else {
+    const seed = new Seed();
+    console.log(seed.toString())
+}
+
+//from empty seed
+//const seed = new Seed();
+//console.log(seed.toString());
+
+//from hex formatted seed
+//const hexSeed = "0x21edc3dec3ef1873cf8f333381c5f360c3"
+//const seed = new Seed(hexSeed);
+//console.log(seed.toString());
+
+//from rippled RFC1751 mnemonic
+//const rippledMnemonic = 'KANT FISH GENE COST WEB JAKE CHUM HAD SOD MID KICK BOTH'
+//const seed = new Seed(rippledMnemonic);
+//console.log(seed.toString());
```

Comment:
I would remove the fromMnemonic part of this implementation as it's a deprecated feature of rippled to generate these mnemonics, it's only been added to the library to have backwards compatibility with new accounts generated using the old `wallet_propose` method, but we're planning on deprecating the client library support for it.

Reply:
It is gone, that this is deprecated somehow slipped me